libraries/security_group.rb

```diff
@@ -0,0 +1,36 @@
+class Hash
+  # Removes empty values from a hash
+  def remove_empty
```

Comment:
I think this can be more clearly written. Here is an (untested) example inspired from rails source:
```ruby
# Returns a true if an object is nil or empty
#   empty_array = []
#   empty_hash = {}
#   array = [0,1,2]
#   hash = { one: 1, two: 2, tree: 3}
#   blank?(empty_array) => true
#   blank?(empty_hash} => true
#   blank?(array) => false
#   blank?(hash) => false
#   hash.each_pair do |k,v|
#       blank?(k) || blank?(v) => false
#   end
#   empty_hash.each_pair do |k,v|
#       blank?(k) || blank?(v) => true
#   end
def blank?(obj)
    return true if obj.nil? || obj.respond_to(:'empty?') && obj.empty?
    false
end

# Replaces the hash without the empty key/values.
#   hash = { a: true, b: false, c: nil}
#   array = [0, '1', nil]
#   remove_blank!(hash) # => { a: true, b: false}
#   hash # => { a: true, b: false }
#   remove_blank!(array) # => [0, '1']
#   array # => [0, '1']
def remove_blanks!(obj)
    case obj.class
    when 'Hash'
        obj.each_pair do |k, v|
            delete(key) if blank?(k) || blank?(v)
        end
    when 'Array'
        obj.each do |elem|
            delete(elem) if blank?(elem)
        end
    end
    obj      
end
```

Reply:
I tried those and several variations without any luck.  what's the objection with what is in the PR?  that it is patching the Hash class?  lack of comments?  It's 8 lines of code so very small and the usage is very limited - I wasn't too concerned with it however I can make changes if necessary.

Reply:
No real objection just thought it could be simpler for someone who is not very familiar with ruby development to read. Using `&block` and `*splat` operators are not intuitive to new ruby developers.